reconcile.go

```diff
@@ -85,7 +87,45 @@ func useProviderToTalkToAzure(provider *plugin.GRPCProvider) {
 		log.Println(configureProviderResp.Diagnostics.Err().Error())
 		panic("Failed to configure provider")
 	}
+}
+
+func reconcileCrd(provider *plugin.GRPCProvider, kind string, crd *unstructured.Unstructured) {
+	// Get the kinds terraform schema
+	resourceName := "azurerm_" + strings.Replace(kind, "-", "_", -1)
+	rgSchema := provider.GetSchema().ResourceTypes[resourceName]
+
+	configValue := createEmptyResourceValue(rgSchema, "test1")
+
+	// Get the spec from the CRD
+	spec := crd.Object["spec"].(map[string]interface{})
+	jsonSpecRaw, err := json.Marshal(spec)
+	if err != nil {
+		panic(err)
+	}
+
+	// Create a TF cty.Value from the Spec JSON
+	configValue, err = applyValuesFromJSON(rgSchema, configValue, string(jsonSpecRaw))
+	if err != nil {
+		panic(err)
+	}
+
+	// Either update or create the resource
+	var newState []byte
+	state, exists := crd.GetAnnotations()["tfstate"]
+	if exists {
+		// Updating existing object
+		newState = planAndApplyConfig(provider, resourceName, *configValue, []byte(state))
+	} else {
+		// Creating a new object
+		newState = planAndApplyConfig(provider, resourceName, *configValue, []byte{})
+	}
+
+	//Todo: persist state
+	_ = newState
```

Comment:
Is it worth including setting the `tfstate` annotation in this PR, or were you planning to leave that for a future PR?

Reply:
Cool, later PR works for me